tests/test_matrix.py

```diff
@@ -312,6 +312,46 @@ def test_determinant():
         assert a == "{1.0}"
 
 
+def test_inverse():
+    mat = getTokens("[5, 7, 9;\
+                    4, 3, 8;\
+                    7, 5, 6]")
+    if mat.isSquare():
+        a = ''
+        for i in mat.determinant():
```

Comment:
You can use the `tokensToString()` from `visma.io.parser` to make the zero check.

Reply:
@8hantanu  Will it be better to just move the check for the singular matrix inside the inverse function itself?

Reply:
Yes. That would actually be better.